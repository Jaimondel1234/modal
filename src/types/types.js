export const types = {
  authLogin: "[Auth] Login",
  authLogout: "[Auth] Logout",

  uiOpenModal: "[ui] Open Modal",
  uiCloseModal: "[ui] Close Modal",

  noteAddNew: "[notes] Add New Notes",
  noteDeleted: "[notes] Delete Note",
  noteSetActive: "[notes] Set Active Note",
  noteUpdate: "[notes] Update note",
  noteSetListMode: "[notes] Set List Mode ",
  noteReset: "[notes] Reset notes",
};
