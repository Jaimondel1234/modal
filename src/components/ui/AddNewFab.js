import React from "react";
import { faPlus } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

import styled from "styled-components";
import { useDispatch } from "react-redux";
import { uiOpenModal } from "../../actions/ui";
import { FabStyle } from "../../styles/core";

const PlusFabStyle = styled(FabStyle)`
  border-radius: 100%;
  font-size: 50px;
  padding: 30px;
  position: fixed;
  right: 40px;
  bottom: 25px;
  color: white;
  background-color: #007bff;
`;

/*
  Button to Open the modal
*/
export const AddNewFab = () => {
  const dispatch = useDispatch();

  const handleClick = () => {
    dispatch(uiOpenModal());
  };
  return (
    <PlusFabStyle onClick={handleClick}>
      <FontAwesomeIcon icon={faPlus} />
    </PlusFabStyle>
  );
};
