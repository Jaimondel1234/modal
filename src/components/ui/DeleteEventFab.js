import React from "react";
import PropTypes from "prop-types";
import { faTrash } from "@fortawesome/free-solid-svg-icons/faTrash";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { useDispatch } from "react-redux";
import styled from "styled-components";
import { noteDelete } from "../../actions/note";
import { FabStyle } from "../../styles/core";

const TrashFabStyle = styled(FabStyle)`
  left: 8%;
  bottom: 75%;
`;

/*
  Button to delete a note
*/
export const DeleteEventFab = ({ noteId }) => {
  const dispatch = useDispatch();

  const handleClick = () => {
    dispatch(noteDelete(noteId));
  };

  return (
    <TrashFabStyle onClick={handleClick}>
      <FontAwesomeIcon icon={faTrash} />
    </TrashFabStyle>
  );
};

DeleteEventFab.propTypes = {
  noteId: PropTypes.number.isRequired,
};
