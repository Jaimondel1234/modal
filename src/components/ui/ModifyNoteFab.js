import React from 'react';
import PropTypes from 'prop-types';
import { faPencilAlt } from '@fortawesome/free-solid-svg-icons/faPencilAlt';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { useDispatch } from 'react-redux';
import styled from 'styled-components';
import { noteSetActive } from '../../actions/note';
import { uiOpenModal } from '../../actions/ui';
import { FabStyle } from '../../styles/core';

const PencilAltFabStyle = styled(FabStyle)`
  right: 8%;
  bottom: 75%;
`;

/*
  Button to modify a note
*/
export const ModifyNoteFab = ({ note }) => {
  const dispatch = useDispatch();
  const handleClick = () => {
    dispatch(noteSetActive(note));
    dispatch(uiOpenModal());
  };

  return (
    <PencilAltFabStyle onClick={handleClick}>
      <FontAwesomeIcon icon={faPencilAlt} />
    </PencilAltFabStyle>
  );
};

ModifyNoteFab.propTypes = {
  note: PropTypes.shape({
    id: PropTypes.number,
    title: PropTypes.string,
    imgUrl: PropTypes.string,
    description: PropTypes.string,
  }),
};
