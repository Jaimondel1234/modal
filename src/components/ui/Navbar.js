import React from 'react';
import { Link } from 'react-router-dom';
import { NavbarList, NavbarElement, ButtonStyle } from '../../styles/core';
import styled from 'styled-components';
import { useDispatch } from 'react-redux';
import { noteReset } from '../../actions/note';
import { authLogout } from '../../actions/auth';

const ButtonLogoutStyle = styled(ButtonStyle)`
  display: block;
  font-weight: bold;
  font-size: 40px;
  text-align: center;
  letter-spacing: 5px;
  line-height: 60px;
`;

/* 
   Navbar
 */

export const Navbar = () => {
  const dispatch = useDispatch();

  const handleLogout = () => {
    localStorage.removeItem('user');
    localStorage.removeItem('notes');
    dispatch(authLogout());
    dispatch(noteReset());
  };
  return (
    <>
      <NavbarList>
        <NavbarElement>
          <Link to='/home'>Home</Link>
        </NavbarElement>
        <NavbarElement>
          <Link to='/notes'>Notes</Link>
        </NavbarElement>
        <NavbarElement float='right'>
          <ButtonLogoutStyle onClick={handleLogout} m='5px' p='5px 10px'>
            Logout
          </ButtonLogoutStyle>
        </NavbarElement>
      </NavbarList>
    </>
  );
};
