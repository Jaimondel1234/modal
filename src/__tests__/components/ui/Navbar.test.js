import React from 'react';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { Navbar } from '../../../components/ui/Navbar';
import { MemoryRouter } from 'react-router-dom';
import { types } from '../../../types/types';

describe('Test <Navbar/>', () => {
  Storage.prototype.removeItem = jest.fn();
  const mockStore = configureStore([]);
  const initState = {};

  const store = mockStore(initState);
  store.dispatch = jest.fn();
  const wrapper = mount(
    <Provider store={store}>
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>
    </Provider>,
  );
  test('it must be displayed correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('if the button is clicked, the logout action must be dispatched', () => {
    wrapper.find('button').prop('onClick')();
    expect(localStorage.removeItem).toHaveBeenNthCalledWith(1, 'user');
    expect(localStorage.removeItem).toHaveBeenNthCalledWith(2, 'notes');
    expect(store.dispatch).toHaveBeenNthCalledWith(1, {
      type: types.authLogout,
    });
    expect(store.dispatch).toHaveBeenNthCalledWith(2, {
      type: types.noteReset,
    });
  });
});
