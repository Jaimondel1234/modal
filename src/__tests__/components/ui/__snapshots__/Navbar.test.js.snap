// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Navbar/> it must be displayed correctly 1`] = `
<ul
  className="core__NavbarList-sc-1mkz20q-7 giEOfX"
>
  <li
    className="core__NavbarElement-sc-1mkz20q-8 cfUFXr"
  >
    <a
      href="/home"
      onClick={[Function]}
    >
      Home
    </a>
  </li>
  <li
    className="core__NavbarElement-sc-1mkz20q-8 cfUFXr"
  >
    <a
      href="/notes"
      onClick={[Function]}
    >
      Notes
    </a>
  </li>
  <li
    className="core__NavbarElement-sc-1mkz20q-8 fMsjzQ"
  >
    <button
      className="core__ButtonStyle-sc-1mkz20q-5 Navbar__ButtonLogoutStyle-x8fbn3-0 jVbggV dLcfzz"
      onClick={[Function]}
    >
      Logout
    </button>
  </li>
</ul>
`;
